godal.cpp

```diff
@@ -1766,5 +1766,71 @@ GDALDatasetH godalNearblack(cctx *ctx, const char *pszDest, GDALDatasetH hDstDS,
 	}
 
 	godalUnwrap();
+	return ret;
+}
+
+OGRSpatialReferenceH godalGetGCPSpatialRef(GDALDatasetH hSrcDS) {
+	return GDALGetGCPSpatialRef(hSrcDS);
+}
+
+int godalGetGCPCount(GDALDatasetH hSrcDS) {
+	return GDALGetGCPCount(hSrcDS);
+}
+
+const GDAL_GCP *godalGetGCPs(GDALDatasetH hSrcDS) {
+	return GDALGetGCPs(hSrcDS);
+}
+
+const char *godalGetGCPProjection(GDALDatasetH hSrcDS) {
+	return GDALGetGCPProjection(hSrcDS);
+}
+
+void godalSetGCPs(cctx *ctx, GDALDatasetH hSrcDS, int numGCPs, GDAL_GCP *GCPList, const char *pszGCPProjection) {
+	godalWrap(ctx);
+
+	CPLErr ret = GDALSetGCPs(hSrcDS, numGCPs, GCPList, pszGCPProjection);
+	if(ret!=0) {
+		forceCPLError(ctx, ret);
+	}
+
+	GDALDeinitGCPs(numGCPs, GCPList);
+	CPLFree(GCPList);
+	
+	godalUnwrap();
+	return;
+}
+
+void godalSetGCPs2(cctx *ctx, GDALDatasetH hSrcDS, int numGCPs, GDAL_GCP *GCPList, OGRSpatialReferenceH hSRS) {
+	godalWrap(ctx);
+
+	CPLErr ret = GDALSetGCPs2(hSrcDS, numGCPs, GCPList, hSRS);
+	if(ret!=0) {
+		forceCPLError(ctx, ret);
+	}
+
+	GDALDeinitGCPs(numGCPs, GCPList);
+	CPLFree(GCPList);
+	
+	godalUnwrap();
+	return;
+}
+
+// Utility function to aid conversion from Go `GCP` type to `GDAL_GCP*`, allocates memory for `GDAL_GCP*` and requires a manual `CPLFree`
+GDAL_GCP *godalGCPPropertiesToGDALGCP(int numGCPs, char **pszIds, char**pszInfos, double *dfGCPPixels, double *dfGCPLines, double *dfGCPXs, double *dfGCPYs, double *dfGCPZs) {
+	GDAL_GCP *ret = static_cast<GDAL_GCP *>(CPLCalloc(numGCPs, sizeof(GDAL_GCP)));
+
+    GDALInitGCPs(numGCPs, ret);
+
+    for (int i = 0; i < numGCPs; i++)
+    {
+		ret[i].pszId = CPLStrdup(pszIds[i]); 
+		ret[i].pszInfo = CPLStrdup(pszInfos[i]);
```

Comment:
I believe that the corresponding CPLFrees are never called here?

Reply:
The CPLFrees are called in `godalSetGCPs` and `godalSetGCPs2`, when they call `GDALDeinitGCPs`. I agree with your later comment about calling `GDALInitGCPs` and `GDALDeInitGCPs` in the same function. I will work on reworking `godalSetGCPs` shortly, which should address this.